scripts/setup.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh -ex
+export PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+export DEBIAN_FRONTEND=noninteractive
+export ETCKEEPER_NO_PUSH=1
+
+die() {
+    echo "$@" >&2
+    exit 1
+}
+
+in_docker() {
+    [ -f /.dockerenv ]
+}
+
+# Some basic assumptions
+apt update
+apt install -y apt-transport-https etckeeper
+
+if [ -d /etc-git/ ]; then
+    [ -d /etc-git/.git ] || die "/etc-git is not a git repo, something has gone wrong"
+else
+    # Assume we are pulling the current production branch
+    git clone https://github.com/hashbang/shell-etc.git /etc-git
+fi
+
+
+# Fix some things that maintainer scripts expect
+# Those should probably be filed as Debian bugs...
+mkdir -p /var/lib/bitlbee
+mkdir -p /usr/local/lib/luarocks/rocks-5.1 #WTF is that even needed?
+apt install -y gawk # gawk is required for python2.7-minimal
+
+
+# Make sure the root can git-commit
+# This can happen when pkg updates result in conffile changes
+git config --global user.name  "Enoch Root"
+git config --global user.email "root@hashbang.sh"
+
+
+# Deal with Docker quirks
+if in_docker; then
+    echo 'resolvconf resolvconf/linkify-resolvconf boolean false' | \
+	debconf-set-selections
+fi
+
+
+# Apply the new config
+# It is done this way to avoid removing things that are in .gitignore
+rm -rf /etc/.git
+rsync -aAX /etc-git/ /etc/
+rm -rf /etc-git
+git -C /etc clean -df
+
+# Restore correct permissions in /etc
+# This requires having the correct /etc/passwd and /etc/group,
+# which got copied over in the previous step
+etckeeper init
+
+# Update the apt and dpkg caches
+apt-get update
+apt-cache dumpavail | dpkg --update-avail
+
+# Set dpkg selections
+dpkg --clear-selections
+dpkg --set-selections < /etc/packages.txt
+
+# Install packages, keep the configuration from git
+apt-get dselect-upgrade -o Dpkg::Options::="--force-confold" -q -y --force-yes
```

Comment:
@dpflug For hosted-for-free, integrate-with-Github tools, I know that [Reviewable](https://reviewable.io) is quite good (and could be a solution for the here-and-now).

For something we can selfhost, I heard many good things about [Phabricator](https://phacility.com/phabricator/):
- good CLI tools;
- supports custom rules, for instance stuff like “admins can merge and deploy signed changes without review, and those changes are flagged for later audit”, so hotfixes can be deployed immediately, and still someone else will eventually look at them.

@RyanSquared As far as I can tell, Gitea has no code-review support (not even leaving comments on pull-requests).

If we need further discussion on this, please open an issue under hashbang/hashbang